lms/views/api/vitalsource.py

```diff
@@ -1,32 +1,37 @@
+from pyramid.httpexceptions import HTTPNotFound
 from pyramid.view import view_config, view_defaults
 
 from lms.security import Permissions
 from lms.services.exceptions import ProxyAPIError
-from lms.views.api.vitalsource_sample_data import book_data, toc_data
 
 
 @view_defaults(renderer="json", permission=Permissions.API)
 class VitalSourceAPIViews:
     def __init__(self, request):
         self.request = request
+        self.vitalsource_service = request.find_service(name="vitalsource")
 
     @view_config(route_name="vitalsource_api.books.info")
     def book_info(self):
         book_id = self.request.matchdict["book_id"]
 
-        # In future this will fetch data from VitalSource.
-        if book_id not in book_data:
-            raise ProxyAPIError(f"Book {book_id} not found")
+        book_info = self.vitalsource_service.book_info(book_id)
 
-        return book_data[book_id]
+        if not book_info:
+            raise HTTPNotFound()
+
+        return {
```

Comment:
I think this is what we settled for on BB, the schema represents more or less the third party responses, we adapt it to our client on the view.

Reply:
Yep, this is the same as what Bb does